### ISSUE TITLE
Headers entries is parsed as response body 

### ISSUE THREAD

**Issue by xly0713** [pintsized/lua-resty-http#118]
![image](https://user-images.githubusercontent.com/4737280/28924241-9ddd58f0-7893-11e7-8353-489c58429ab4.png)

The "X-Pase-SvCdn-Ip" header and the following contents will be parsed as response body, this is wrong.

**Comment by pintsized** [comment]
This is because your `X-Parse-Client-Ip` header field has no value, which has no proper meaning, but causes the parsing to fail. I agree that's a bug (it should be more robust), but you can perhaps fix it yourself by not sending an invalid empty response header field (just a single white space character would solve it).

I'll fix it shortly.

**Comment by kleisauke** [comment]
I've also encountered this bug when requesting this [uri](https://raw.githubusercontent.com/recurser/exif-orientation-examples/master/Portrait_6.jpg). For example:

```lua
local http = require "resty.http"
local httpc = http.new()
local res, err = httpc:request_uri("https://raw.githubusercontent.com/recurser/exif-orientation-examples/master/Portrait_6.jpg", {
    method = "GET",
    headers = {
        ['Accept-Encoding'] = 'gzip',
        ['User-Agent'] = 'Mozilla/5.0 (compatible; ImageFetcher/8.0; +http://images.weserv.nl/)'
    },
    ssl_verify = false
})

if not res then
    ngx.say("failed to request: ", err)
    return
end

ngx.say(res.body)
```

This is because the `X-Geo-Block-List` header is empty, see for the REDbot output [here](https://redbot.org/?uri=https%3A%2F%2Fraw.githubusercontent.com%2Frecurser%2Fexif-orientation-examples%2Fmaster%2FPortrait_6.jpg).
![redbot](https://user-images.githubusercontent.com/12746591/43047272-03e6ef7e-8dd5-11e8-9ecc-f7c9b534980a.png)

This patch will fix it (on [line 341](https://github.com/pintsized/lua-resty-http/blob/master/lib/resty/http.lua#L341)):
```diff
diff --git a/lib/resty/http.lua b/lib/resty/http.lua
index 1111111..2222222 100644
--- a/lib/resty/http.lua
+++ b/lib/resty/http.lua
@@ -338,7 +338,7 @@
             return nil, err
         end
 
-        local m, err = ngx_re_match(line, "([^:\\s]+):\\s*(.+)", "jo")
+        local m, err = ngx_re_match(line, "([^:\\s]+):\\s*(.*)", "jo")
         if not m then
             break
         end
```

@pintsized Should I make a PR with this fix?

**Comment by hamishforbes** [comment]
@kleisauke Yes please! A PR would be helpful, especially if you can include a test that breaks this in the current version

**Comment by kleisauke** [comment]
A pull request has been made to fix this bug at #151.
